packages/matter.js/src/protocol/interaction/SubscriptionHandler.ts

```diff
@@ -360,6 +363,30 @@ export class SubscriptionHandler {
             });
         }
 
+        const { newEvents } = this.registerNewEvents();
+        newEvents
+            .flatMap(({ path, event: { schema } }): EventPathWithEventData<any>[] => {
+                // But we use eventFilters because we do not want to send all events to the controller
+                const matchingEvents = this.eventHandler.getEvents(path, this.eventFilters);
+                return matchingEvents.map(event => ({
+                    schema,
+                    path,
+                    event,
+                }));
+            })
+            .sort((a, b) => {
```

Comment:
I think you could write as `.sort((a, b) => (a.event?.eventNumber ?? 0) - (b.event?.eventNumber ?? 0)))`

Reply:
Nope ... nort expects a number as result but eventNumber normally is a Bigint ... I will fix some stuff around that mixtures here